### ISSUE TITLE
CA1508 false positive in IAsyncEnumerable.ConfigureAwait

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn-analyzers#4327]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-rc2](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

CA1508 fired on implicit lowered code of `await foreach`.

### Steps To Reproduce

Using `<AnalysisMode>AllEnabledByDefault</AnalysisMode>`:

```csharp
        static async IAsyncEnumerable<int> M(IAsyncEnumerable<int> iae)
        {
            await foreach (int i in iae.ConfigureAwait(false)) // here
            {
                if (i > 0)
                    yield return i;
            }
        }
```

### Expected behavior

Not fired. The comparison with null might come from defensive pattern in compiler lowered code for `await foreach`.

### Actual behavior

Warning CA1508: `iae.ConfigureAwait(false)` can never be `null`.

**Comment by JSkimming** [comment]
I get the same when using `WithCancellation()` I suspect the current fix will suffice, but you may want to add an additional test case.
